pkgs/build-support/emacs/wrapper.nix

```
@@ -196,12 +197,18 @@ runCommand
     # Wrap emacs and friends so they find our site-start.el before the original.
     for prog in $emacs/bin/*; do # */
       local progname=$(basename "$prog")
+      local autoloadExpression=""
       rm -f "$out/bin/$progname"
+      if [[ $progname == emacs ]]; then
```

```suggestion
      if [[ "$progname" == "emacs" ]]; then
```

Sorry I had edited my earlier comment but guess you didn't see the latest edit. Not a sh/bash expert but I think this is more correct (and it's what I tested).

---

See my answer/question above.